tests/pyright/decorators.py

```diff
@@ -88,6 +90,58 @@ def valid_method_no_info(self) -> Self: ...
     def valid_method_info(self, info: ValidationInfo) -> Self: ...
 
 
+class BeforeFieldValidator(BaseModel):
+    """Same tests should apply to `mode='plain'`."""
+
+    @field_validator('foo', mode='before')
+    def no_classmethod(self, value: Any) -> Any:
+        """TODO this shouldn't be valid, the decorator should only work on classmethods.
+
+        We might want to do the same type checking as wrap model validators.
+        """
+
+    @field_validator('foo', mode='before')
+    @classmethod
+    def valid_classmethod(cls, value: Any) -> Any: ...
+
+    @field_validator('foo', mode='before')  # pyright: ignore[reportArgumentType]
+    @classmethod
+    def invalid_with_info(cls, value: Any, info: int) -> Any: ...
+
+    @field_validator('foo', mode='before', input_type=int)  # `input_type` allowed here.
+    @classmethod
+    def valid_with_info(cls, value: Any, info: ValidationInfo) -> Any: ...
```

Comment:
You mention in the PR description:

> want to enforce the validator functions to check for any type (by using typing.Any)

Could you elaborate on this?

Reply:
Because in reality any value can be passed in, so the type hint won't reflect the actual possible values. I prefer to enforce that users use `Any` for the `value` type annotation, so that they make sure to actually check the allowed values. If we were to encourage users to annotate `value` to the allowed types:

```python
@field_validator('foo', mode='before') 
    @classmethod
    def valid_with_info(cls, value: int, info: ValidationInfo) -> Any: ...
```

Users will likely forget to do the proper runtime checks on `value` and assume it is already an `int`. This is even more likely if they do something like:

```python
@field_validator('foo', mode='before') 
    @classmethod
    def valid_with_info(cls, value: int, info: ValidationInfo) -> Any:
        if not isinstance(value, int):  # This should be done by users
            raise ValidationError(...)  # Type checker error, marked as unreachable code
```

Reply:
I agree with @Viicos here, it's dangerous to annotate the field as accepting anything but `Any`/`object` (technically `object` is the safer thing to use as the type hint for this, and then do `isinstance` checks to narrow it), as it will mean that the type checker will assume that you can't get arbitrary stuff in the before validator, but you can.